// angular_reactor/src/core.js

// Imports Supporting Classes

import Router         from 'ARRouter';
import Controller     from 'ARController';
import Model          from 'ARModel';

/**
 * @class The AR class is the primary exportable Angular Reactor core which 
 * handles application setup including controller, models and routing.
 */
default export class AR {

  /**
   * Sets up the inital application
   * @type {String}     The application name
   */
  constructor (applicationName = 'default') {
  
    this.applicationName    = applicationName;
    this.controllers        = [];
    this.models             = [];
    this.routes             = {};

  }

  /**
   * Registers the inital application controllers
   * @type {Array}    The controllers to be intitally setup for the application
   */
  initControllers (controllers = []) {
    this.controllers = controllers;    
  }

  /**
   * Registers the inital application models
   * @type {Array}    The models to be intitally setup for the application
   */
  initModels (models = []) {
    this.models = models;    
  }

  /**
   * Registers the inital application routes
   * @type {Array}    The routes to be intitally setup for the application
   */
  initRoutes (routes = {}) {
    this.routes = routes;    
  }


}
